modules/javafx.graphics/src/main/native-glass/monocle/egl/egl_ext.h

```diff
@@ -25,22 +25,44 @@
 #ifndef __EGL_EXT__
 #define __EGL_EXT__
 #include <jni.h>
+
+// This header file declares functions that need to be provided by low-level
+// drivers or libraries.
+
+// get a handle to the native window (without specifying what window is)
 extern jlong getNativeWindowHandle(const char *v);
+
+// get a handle to the EGL display
 extern jlong getEglDisplayHandle();
+
+// initialize the EGL system with the specified handle
 extern jboolean doEglInitialize(void* handle);
```

Comment:
Is it possible to declare arguments as `jlong eglDisplay` here instead of `void* handle` as it is done in other methods?
As I see nEglInitialize() passes eglDisplay on the java side:
https://github.com/openjdk/jfx/blob/47700d8ef0175d4b457bb658371d2da4ec0a8181/modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/EGLAcceleratedScreen.java#L48
but it needs to convert it to a ptr in the native code:
https://github.com/openjdk/jfx/blob/47700d8ef0175d4b457bb658371d2da4ec0a8181/modules/javafx.graphics/src/main/native-glass/monocle/egl/eglBridge.c#L46

Reply:
What would be the advantage of changing it? In any case, it seems unrelated to this fix.

Reply:
All other methods doEglChooseConfig, doEglCreateWindowSurface, doEglCreateContext, doEglMakeCurrent, doEglSwapBuffers use `jlong eglDisplay`. It would be good if all functions have the same style in function argument declaration.

I am wondering, was it made by purpose and `void* handle` really make sense in doEglInitialize function?
If no, I can change it in a separate fix.

Reply:
I agree it would be nice to have the same style in all functions, but in that case I think a `void*` is more appropriate, as that is what `EGLDisplay` is (https://www.khronos.org/registry/EGL/api/1.1/EGL/egl.h)
One of the goals of `egl_ext.h` file is to stay as close as possible to the EGL specifications, avoiding the need for cast operations in the low-level implementation. Hence, it would make sense to completely get rid of the `jlong` declarations at all. 
It is the task of the `eglBridge.c` file to convert from Java style into EGL-style and vice versa. 

I am willing to change this, but then a separate issue would be more appropriate I believe.